posthog-core/src/index.ts

```diff
@@ -363,8 +429,8 @@ export abstract class PostHogCore {
     const featureFlags = this.getFeatureFlags()
 
     if (!featureFlags) {
-      // If we haven't loaded flags yet we respond undefined to indicate this
-      return undefined
+      // If we haven't loaded flags yet, or errored out, we respond with the default result
+      return defaultResult
```

Comment:
This was intentionally left as undefined. Otherwise there is no way for the client to know whether they are getting the actual loaded flag or just the default.

Not sure what the benefit of returning the defaultResult is given that the client could just do `posthog.getFeatureFlag(foo) ?? myDefaultResult`.

Reply:
I already coded against this change 🙈

Reply:
Hmm, was thinking about this, and wonder if its best to remove the `defaultResult` from the API interface.

Exactly because of what you say -> the defaultResult in the API isn't doing much at all.

For where I'm coming from, consider the case where `defaultResult` is `true`, this messes up any place we had `|| defaultResult`, as it turns a false valid reply into true.

---

@weyert as in the existing change?

Reply:
Weyert is trialing this lib using the alpha version hence his comment.

I think removing it makes total sense as we're basically replicating native functionality in our API for no good reason

Reply:
Yes, what @benjackwhite. I am trying out the alpha of the Node package and updated my OpenFeature provider to correctly indicate missing feature flag when `undefined` is returned.

Reply:
I see, sounds good, the behaviour is consistent then in this version.